### PR TITLE
Enable caching when the mime type is `text/javascript`

### PR DESCRIPTION
**Pull request by trajano**
https://github.com/RMerl/asuswrt-merlin.ng/issues/657

### CODE REVIEW

File: release/src/router/httpd/httpd.c [hunk 1]
```diff
@@ -589,7 +595,7 @@ send_headers( int status, char* title, char* extra_header, char* mime_type, int
     (void) fprintf( conn_fp, "x-frame-options: SAMEORIGIN\r\n");
     (void) fprintf( conn_fp, "x-xss-protection: 1; mode=block\r\n");
 
-    if (fromapp != 0){
+    if (fromapp != 0 && !check_mime_type_and_return_true_if_javascript(mime_type)){
```
Comment (paldier) [inline release/src/router/httpd/httpd.c:598]: `enum {
	FROM_BROWSER,
	FROM_ASUSROUTER,
	FROM_DUTUtil,
	FROM_ASSIA,
	FROM_IFTTT,
	FROM_ALEXA,
	FROM_WebView,
	FROM_UNKNOWN
};`
`if (fromapp != 0)` this means the request is not from the browser, you can't clear the cache on these devices until you reboot these devices.
`struct mime_handler mime_handlers[]` you can find what you want.

FROM_BROWSER:web browser,such as chrome, firefox, ie
FROM_ASUSROUTER:asus app
FROM_DUTUtil:fcc testing tool or development tool
FROM_ASSIA/FROM_IFTTT/FROM_ALEXA:iot device
FROM_WebView:asus app web browser

**Comment by trajano** [comment]
I'm thinking I just want to simplify it to just a few files specifically the jquery-ui and require.js since those are minimized I don't think it will have any place holder values.  But I am not passed the file name on this method, if I can somehow pass it up to here I can probably tweak it.

**Comment by paldier** [comment]
don't change anything in httpd.c

**Comment by RMerl** [comment]
Javascripts can not be blindly cached, they contain dynamically generated data.

**Pull request closed without merging**